### PR TITLE
[AIRFLOW-4846] Allow specification of an existing secret containing git credentials for init containers

### PR DESCRIPTION
**Pull request by george-miller**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4846
  
### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

I would like to specify username/password git credentials as a secret so that I don't have plaintext username/passwords stored in my airflow configuration file.  We can assume the given secret has GIT_USERNAME and GIT_PASSWORD fields.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -678,6 +678,9 @@ git_dags_folder_mount_point =
 git_ssh_key_secret_name =
 git_ssh_known_hosts_configmap_name =
 
+# To give git_sync a secret containing git credentials, use this
```
Comment (ashb) [inline airflow/config_templates/default_airflow.cfg]: Could you extend the comment to say what fields are read from the secret?

**Review by ashb — CHANGES_REQUESTED** [review]
Looks good - one small change to make this easier for people to know how to use.

File: airflow/config_templates/default_airflow.cfg (hunk 1 above)
Comment (george-miller) [reply airflow/config_templates/default_airflow.cfg]: Definitely.  How does this look @ashb ?

**Comment by george-miller** [comment]
@ashb Sorry to bother you.  Mind looking at this again?  Thanks!

**Review by ashb — APPROVED** [review]

**Pull request merged**